src/viam/examples/modules/complex/test_complex_module.cpp

```diff
@@ -69,60 +70,25 @@ BOOST_AUTO_TEST_SUITE_END()
 
 BOOST_AUTO_TEST_SUITE(test_gizmo_client_server)
 
-// This sets up the following architecture
-// -- MockComponent
-//        /\
-//
-//        | (function calls)
-//
-//        \/
-// -- ComponentServer (Real)
-//        /\
-//
-//        | (grpc InProcessChannel)
-//
-//        \/
-// -- ComponentClient (Real)
-//
-// This is as close to a real setup as we can get
-// without starting another process
-//
-// The passed in lambda function has access to the ComponentClient
-template <typename Lambda>
-void gizmo_server_to_mock_pipeline(Lambda&& func) {
-    GizmoServer gizmo_server;
-    gizmo_server.resource_manager()->add(std::string("testgizmo"), test_gizmo_impl::get_gizmo());
-
-    grpc::ServerBuilder builder;
-    builder.RegisterService(&gizmo_server);
-
-    std::unique_ptr<grpc::Server> server = builder.BuildAndStart();
-
-    grpc::ChannelArguments args;
-    auto grpc_channel = server->InProcessChannel(args);
-    GizmoClient client("testgizmo", grpc_channel);
-    // Run the passed test on the created stack
-    std::forward<Lambda>(func)(client);
-    // shutdown afterwards
-    server->Shutdown();
-}
-
 BOOST_AUTO_TEST_CASE(test_do_one) {
-    gizmo_server_to_mock_pipeline([](Gizmo& client) -> void {
+    const auto mock = test_gizmo_impl::get_gizmo();
+    client_to_mock_pipeline<GizmoClient, GizmoServer>(mock, [](Gizmo& client) {
```

Comment:
The new `client_to_mock_pipeline` requires specifying your `Client` and `Server` types and passing in your `mock` resource. The second argument lambda has access to the created client, and the mock can be accessed if needed by capturing by reference (`[&]`, I tried to use `[]` when not necessary)